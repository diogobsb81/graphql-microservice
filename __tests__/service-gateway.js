// https://blog.apollographql.com/4-simple-ways-to-call-a-graphql-api-a6807bcdb355
require("isomorphic-fetch");

const query = `{
  allPosts {
    id
    title
    author {
      fullName
      email
      posts {
        title
      }
    }
  }
  allUsers {
    id
    posts {
      title
    }
  }
  post(id: 1) {
    title
  }
}`;

test("gateway should response to query", async () => {
  await fetch("http://localhost:3000/", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ query })
  })
    .then(res => res.json())
    .then(res => {
      expect(res).toMatchSnapshot();
    });
});

test("gateway should answer with identifier", async () => {
  await fetch("http://localhost:3000/", {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({ query: "{ identifier { hash } }" })
  })
    .then(res => res.json())
    .then(res => {
      expect(res.data.identifier.hash).toBeDefined();
    });
});
