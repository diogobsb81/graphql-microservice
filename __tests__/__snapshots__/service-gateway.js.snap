// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gateway should response to query 1`] = `
Object {
  "data": Object {
    "allPosts": Array [
      Object {
        "author": Object {
          "email": "cb@w.eber",
          "fullName": "C.B. Weber",
          "posts": Array [
            Object {
              "title": "Cool first post",
            },
          ],
        },
        "id": 1,
        "title": "Cool first post",
      },
      Object {
        "author": Object {
          "email": "s.baren@d.aren",
          "fullName": "Sara Baren",
          "posts": Array [
            Object {
              "title": "Cool second post",
            },
            Object {
              "title": "Cool third post",
            },
          ],
        },
        "id": 2,
        "title": "Cool second post",
      },
      Object {
        "author": Object {
          "email": "s.baren@d.aren",
          "fullName": "Sara Baren",
          "posts": Array [
            Object {
              "title": "Cool second post",
            },
            Object {
              "title": "Cool third post",
            },
          ],
        },
        "id": 3,
        "title": "Cool third post",
      },
      Object {
        "author": Object {
          "email": "horst@m.eier",
          "fullName": "Horst Meier",
          "posts": Array [
            Object {
              "title": "Cool fourth post",
            },
          ],
        },
        "id": 4,
        "title": "Cool fourth post",
      },
    ],
    "allUsers": Array [
      Object {
        "id": 1,
        "posts": Array [
          Object {
            "title": "Cool first post",
          },
        ],
      },
      Object {
        "id": 2,
        "posts": Array [
          Object {
            "title": "Cool second post",
          },
          Object {
            "title": "Cool third post",
          },
        ],
      },
      Object {
        "id": 3,
        "posts": Array [
          Object {
            "title": "Cool fourth post",
          },
        ],
      },
      Object {
        "id": 4,
        "posts": Array [],
      },
    ],
    "post": Object {
      "title": "Cool first post",
    },
  },
}
`;
